src/plugins/intel_cpu/src/nodes/scaled_attn.cpp

```diff
@@ -806,6 +812,269 @@ bool ScaledDotProductAttention::isSupportedOperation(const std::shared_ptr<const
     return true;
 }
 
+void ScaledDotProductAttention::assignState(const std::shared_ptr<VariableStateKVcache>& state, int idx) {
+    auto inputNumber = getOriginalInputsNumber();
+    if (inputNumber - 2 == static_cast<size_t>(idx)) {
+        m_k_state = state;
+    } else if (inputNumber - 1 == static_cast<size_t>(idx)) {
+        m_v_state = state;
+    } else {
+        OPENVINO_THROW(
+            "Unexpected idx ", idx , " for a state in a node with type: ", getTypeStr(), " and name ", getName());
+    }
+}
+
+void ScaledDotProductAttention::gatherConcatPastkv(const MemoryPtr& mem_cur_k, const MemoryPtr& mem_cur_v, const MemoryPtr& mem_beam_idx) {
+    PlainTensor cur_k;
+    cur_k.reset(mem_cur_k);
+    if (!m_config.config.permute_axes.empty())
+        cur_k = cur_k.permute(m_config.config.permute_axes);
+
+    updateBeamTable(mem_beam_idx, cur_k.size(2));
+    updatePastkv(mem_cur_k, mem_cur_v);
+}
+
+void ScaledDotProductAttention::updateBeamTable(const MemoryPtr& mem_beam_idx, size_t L1) {
```

Comment:
minor: can we add brief explanation what exactly `updateBeamTable` and `updatePastkv` functions are doing?

Reply:
Added.